src/ui/components/Library/Team/View/NewTestRuns/TestRunsContextRoot.tsx

```diff
@@ -53,7 +53,7 @@ export function TestRunsContextRoot({ children }: { children: ReactNode }) {
   const filterByTextDeferred = useDeferredValue(filterByText);
   const [filterTestsByText, setFilterTestsByText] = useState("");
 
-  const [spec, setSpec] = useState<string | null>(null);
+  const [testId, setTestId] = useState<string | null>(null);
```

Comment:
Ah nice — this is going to make it less janky to store the selected test in the URL instead of serializing the path!